src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -105,7 +105,9 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 format!("{} occurs due to use{}", desired_action.as_noun(), use_spans.describe()),
             );
 
-            err.buffer(&mut self.errors_buffer);
+            // Bypass the buffer for this error, so we don't downgrade it to a
+            // warning ever.
+            err.emit();
```

Comment:
The buffer isn't just to downgrade errors, but also to allow sorting them. I would prefer fixing this in the AST borrow checker, although that doesn't appear to be so easy. I would also prefer adding a flag to `MirBorrowckCtxt` to prevent it from downgrading errors.

Reply:
@Centril didn't you recently write a PR that also make NLL errors ignore mogration in some cases  (for match guards)? Can't this use the same approach?

Reply:
@RalfJung I defer to @matthewjasper since I basically used the mechanism they proposed in that PR. :)

Reply:
I added a flag as @matthewjasper suggested, PTAL.